runtime/runtime/src/verifier.rs

```diff
@@ -238,7 +239,8 @@ pub fn verify_and_charge_transaction(
                 )
                 .into());
             }
-            if transaction.receiver_id.as_ref() != function_call_permission.receiver_id {
```

Comment:
You probably don't need `as_ref()` here now at all as there is comparison implementations for AccountIdRef <> str